src/main/java/de/berlin/lostberlin/controller/BusinessController.java

```diff
@@ -40,10 +48,31 @@ public BusinessController(BusinessService businessService) {
     }
 
     @PutMapping("/{id}")
-    public ResponseEntity<Business> updateBusiness(@PathVariable(value = "id") Long id,
-                                         @Valid @RequestBody BusinessUpdateDto businessProfile) {
+    public ResponseEntity<Business> updateBusinessFully(@PathVariable(value = "id") Long id,
+                                                        @Valid @RequestBody BusinessUpdateDto businessProfile) {
 
         return ResponseEntity.status(HttpStatus.OK).body(businessService.saveUpdatedBusinessProfile(id, businessProfile));
     }
 
+    @PatchMapping("/{id}")
+    public ResponseEntity<Business> updateBusinessPartially(@PathVariable(value = "id") Long id,
+                                                            @Valid @RequestBody BusinessUpdatePhotoDto photoFilePath){
+        return ResponseEntity.status(HttpStatus.OK).body(businessService.savePartiallyUpdatedBusinessProfile(id, photoFilePath));
+    }
+
+    @PostMapping("/{id}/uploadFile")
+    public ResponseEntity<String> uploadFile(@RequestParam("file") MultipartFile file) {
+        fileService.storeFile(file);
```

Comment:
Do you handle somehow if file storing has failed?

Reply:
yes, I have a service method storeFile, which throws ResourceNotSavedException.

Reply:
Is this exception then mapped to either 4xx or 5xx when communicated to the client in errornous response?

Reply:
it's mapped to 5xx, @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)